pkg/yttlibrary/url.go

```diff
@@ -178,3 +185,76 @@ func (b urlModule) sortedKeys(vals url.Values) []string {
 	sort.Strings(result)
 	return result
 }
+
+func (b urlModule) ParseURL(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 1 {
+		return starlark.None, fmt.Errorf("expected exactly one argument")
+	}
+
+	urlStr, err := core.NewStarlarkValue(args.Index(0)).AsString()
+	if err != nil {
+		return starlark.None, err
+	}
+
+	parsedURL, err := url.Parse(urlStr)
+	if err != nil {
+		return starlark.None, err
+	}
+
+	return (&urlValue{parsedURL}).AsStarlarkValue(), nil
+}
+
+func (uv *urlValue) AsStarlarkValue() starlark.Value {
+	m := orderedmap.NewMap()
+	m.Set("user", uv.user())
+	m.Set("without_user", starlark.NewBuiltin("url.without_user", core.ErrWrapper(uv.withoutUser)))
+	m.Set("string", starlark.NewBuiltin("url.string", core.ErrWrapper(uv.string)))
+	s := core.NewStarlarkStruct(m)
+	s.SetRepresentation(urlRepresent)
+	return s
+}
+
+func urlRepresent() (starlark.Value, error) {
+	return nil, fmt.Errorf("urlValue cannot be directly referenced, please use urlValue.string()")
```

Comment:
```suggestion
	return nil, fmt.Errorf("urlValue: cannot coerce to a string; use .string()")
```

Attempting to follow the error message formatting I'm seeing elsewhere in Starlark/ytt.